colossalai/tensor/shape_consistency.py

```diff
@@ -403,6 +405,157 @@ def get_all_one_step_transform_spec(self, source_spec: ShardingSpec, orig_cost_d
         valid_spec_dict.update(self.get_all_shard_spec(source_spec, orig_cost_dict))
         return valid_spec_dict
 
+    def mem_cost(self, comm_action_sequence: List[CommSpec]) -> TrainCycleItem:
+        """memory cost of the communication action sequence
+        TODO: Currently we just consider tensor numel in the shape consistency manger,
+        as the manager itself doesn't have the access to tensor dtype, we need to take
+        it into consideration in memory estimation.
+
+        Args:
+            comm_action_sequence (List[CommSpec]): list of communication actions
+
+        Returns:
+            TrainCycleItem: memory cost of such comm_action_sequence
+        """
+
+        def compute_shape(sharding_spec: ShardingSpec):
+            shape = sharding_spec.entire_shape
+            for dim, shard in sharding_spec.dim_partition_dict.items():
+                shape[dim] = shape[dim] // len(shard)
+            return shape
+
+        def gather_analysis(comm_spec: CommSpec, discard_input: bool, alloc_mem: int, peak_mem: int):
```

Comment:
The type of alloc_mem and peak_mem should be float

Reply:
okay~